components/random_cut.py

```diff
@@ -0,0 +1,51 @@
+import numpy as np
+from heapq import heapify, heappush, heappop
+
+
+def random_cut_generator(num_bins, bin_length, bin_width, bin_height, num_cases,
```

Comment:
This and other new public functions and methods should be unit tested as well. One possible unit test for this would be to generate data and check that the total volume is what we expect. This could also use a docstring and type hints on the arguments and return type.